.github/workflows/reusable_goreleaser.yml

```
@@ -0,0 +1,201 @@
+        if: ${{ inputs.gcp_ar_registry_cache_enabled }}
+        # Having params in a given cli as env. vars. does not improve security without extra validation
+        # nosemgrep: yaml.github-actions.security.run-shell-injection.run-shell-injection
+        run: 'gcloud --project ${{ inputs.gcp_ar_registry_cache_project_id }} auth configure-docker us-docker.pkg.dev'
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build in Docker
+        uses: docker/build-push-action@v6
+        if: not ${{ inputs.gcp_ar_registry_cache_enabled }}
+        with:
+          file: ${{ inputs.build_dockerfile }}
+          context: .
+          load: true
+          cache-from: ${{ inputs.gha_cache_enabled && 'type=gha' || '' }}
+          cache-to: ${{ inputs.gha_cache_enabled && 'type=gha,mode=max' || '' }}
+          target: ${{ inputs.build_dockerfile_target }}
+          tags: not-exists.elastiflow.com/builder
+          build-args: |
+            "BINARY=${{ inputs.binary }}"
+            "PKG_NAME=${{ inputs.pkg_name }}"
+            "PKG_FILE_NAME=${{ inputs.pkg_file_name }}"
+            "PKG_DESCRIPTION=${{ inputs.pkg_description }}"
+            ${{ inputs.build_args }}
+          secrets: |
+            "gh_pat_dependencies=${{ secrets.build_gh_pat_dependencies }}"
+            "gpg_key=${{ secrets.build_gpg_key }}"
+            "gpg_passphrase=${{ secrets.build_gpg_passphrase }}"
+      - name: Build in Docker with GCP AR cache
+        uses: docker/build-push-action@v6
+        if: ${{ inputs.gcp_ar_registry_cache_enabled }}
+        with:
+          file: ${{ inputs.build_dockerfile }}
+          context: .
+          load: true
+          cache-from: type=registry,ref=${{ inputs.gcp_ar_registry_cache_name }}
+          cache-to: type=registry,ref=${{ inputs.gcp_ar_registry_cache_name }},mode=max
+          target: ${{ inputs.build_dockerfile_target }}
+          tags: not-exists.elastiflow.com/builder
+          build-args: |
+            "BINARY=${{ inputs.binary }}"
+            "PKG_NAME=${{ inputs.pkg_name }}"
+            "PKG_FILE_NAME=${{ inputs.pkg_file_name }}"
+            "PKG_DESCRIPTION=${{ inputs.pkg_description }}"
+            ${{ inputs.build_args }}
+          secrets: |
+            "gh_pat_dependencies=${{ secrets.build_gh_pat_dependencies }}"
+            "gpg_key=${{ secrets.build_gpg_key }}"
+            "gpg_passphrase=${{ secrets.build_gpg_passphrase }}"
+      - name: Get build artifacts
+        run: |
+          docker run --rm -v `pwd`:/dst --name artifacter not-exists.elastiflow.com/builder sh -c 'cp -R dist /dst/'
+      - name: Publish to APT repository
+        if: ${{ inputs.publish_apt }}
+        # Having params in a given cli as env. vars. does not improve security without extra validation
+        # nosemgrep: yaml.github-actions.security.run-shell-injection.run-shell-injection
+        run: |
+          find dist/ -type f -name '*.deb' | while read pkg_path; do
```

We could use `-print0` and `read -r -d ' to handle filenames in a safer manner. Something like this:
```suggestion
          find dist/ -type f -name '*.deb' -print0 | while IFS= read -r -d '' pkg_path; do
```